File: Mobile/lib/models/receipt_stub.dart
// lib/models/receipt_stub.dart
class ReceiptStub {
  final String? stubID;
  final String? bookID;
  final String? status; // e.g., "pending", "collected"

  ReceiptStub({
    this.stubID,
    this.bookID,
    this.status,
  });

  factory ReceiptStub.fromJson(Map<String, dynamic> json) {
    return ReceiptStub(
      stubID: json['stubID'] as String?,
      bookID: json['bookID'] as String?,
      status: json['status'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'stubID': stubID,
      'bookID': bookID,
      'status': status,
    };
  }
}

File: +/server/TraceFlow-SERVER-HTTPS/Code/Back/models/receipt/receiptStub.js
// models/receipt/receiptStub.js
const { nanoid } = require('nanoid');

module.exports = (sequelize, DataTypes) => {
    return sequelize.define('ReceiptStub', {
        stubID: {
            type: DataTypes.STRING,
            primaryKey: true,
            defaultValue: () => `stub_${nanoid()}`,
        },
        status: {
            type: DataTypes.ENUM('pending', 'collected', 'transmitted', 'archived'),
            allowNull: false,
            defaultValue: 'pending',
        },
        bookID: {
            type: DataTypes.STRING,
            allowNull: false,
            unique: true, // One stub per book
            references: {
                model: 'ReceiptBooks',
                key: 'bookID',
            },
        },
    });
};